app/controllers/admin/programs_controller.rb

```diff
@@ -15,8 +15,14 @@ def update
 
       if @program.update_attributes(program_params)
         ConferenceScheduleUpdateMailJob.perform_later(@conference) if send_mail_on_schedule_public
-        redirect_to admin_conference_program_path(@conference.short_title),
-                    notice: 'The program was successfully updated.'
+        if program_params[:selected_schedule_id].nil?
```

Comment:
Please don't do that, instead use respond_to

```
respond_to do |format|
  format.html { redirect_to(...) }
  format.js
end
```

If you do it like this, you can use it only with js when ``selected_schedule_id`` is present. With respond_to it is more general.

http://apidock.com/rails/ActionController/MimeResponds/InstanceMethods/respond_to